src/garbage_collector.rs

```diff
@@ -129,21 +129,42 @@ struct GarbageCollectorOrchestrator {
     db_stats: Arc<DbStats>,
 }
 
-impl GarbageCollectorOrchestrator {
+trait GcTask {
+    async fn collect(&self, utc_now: DateTime<Utc>) -> Result<(), SlateDBError>;
+    fn status(&mut self) -> &mut DirGcStatus;
+    fn ticker(&self) -> crossbeam_channel::Receiver<Instant>;
+    fn resource(&self) -> &str;
+}
+
+struct ManifestGcTask {
+    manifest_store: Arc<ManifestStore>,
+    db_stats: Arc<DbStats>,
+    manifest_options: Option<GarbageCollectorDirectoryOptions>,
+    status: DirGcStatus,
+}
+
+impl ManifestGcTask {
+    fn new(orchestrator: &GarbageCollectorOrchestrator) -> Self {
```

Comment:
This is kind of nitty, but I generally prefer passing in params directly rather than the whole object (so manifest_store, db_stats, and so on). More verbose, but gives invoking code more flexibility (and easier to unit test).

Reply:
Thanks for the Windsurf rec:
<img width="481" alt="image" src="https://github.com/user-attachments/assets/c02d30af-7a8e-4d46-b751-d0df82d20f23">
Perfect on the first try 🤩